external/dynamo/generalized/router.py

```
@@ -271,17 +302,64 @@ def _iat_factor(iat: str) -> float:
     # --------------------- init --------------------- #
     async def initialize(self):
+        """Initialize router by polling for backend workers."""
         engine = self.runtime.namespace("dynamo").component("backend")
-        logger.info("Getting engine client")
+        logger.info("Getting engine client for dynamo/backend/generate")
         self.engine_client = await engine.endpoint("generate").client()
-        await self.engine_client.wait_for_instances()
+
+        min_workers = int(self.min_workers)
+        if min_workers < 0:
+            raise ValueError(f"min_workers must be >= 0, got {min_workers}")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)